### PR TITLE
AsyncAPI Topics Page revamping

### PR DESCRIPTION
**Pull request by lasanthaS**
This PR contains changes related to the revamping of the AsyncAPI Topics page in the Publisher app.

Fixes https://github.com/wso2/product-apim/issues/10335
Fixes https://github.com/wso2/product-apim/issues/10353
Fixes https://github.com/wso2/product-apim/issues/10354
Fixes https://github.com/wso2/product-apim/issues/10355
Fixes https://github.com/wso2/product-apim/issues/10356
Fixes https://github.com/wso2/product-apim/issues/10357
Fixes https://github.com/wso2/product-apim/issues/10358
Fixes https://github.com/wso2/product-apim/issues/10359

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java [hunk 1]
```diff
@@ -358,10 +358,12 @@ public static API updateApi(API originalAPI, APIDTO apiDtoToUpdate, APIProvider
...
             }
         } else {
-            // TODO: update the asyncapi.yaml
+             String oldDefinition = apiProvider.getAsyncAPIDefinition(apiIdentifier.getUUID(), tenantDomain);
             AsyncApiParser asyncApiParser = new AsyncApiParser();
-            String apiDefinition = asyncApiParser.generateAsyncAPIDefinition(originalAPI);
-            apiProvider.saveAsyncApiDefinition(originalAPI, apiDefinition);
+            String updateAsyncAPIDefinition = asyncApiParser.updateAsyncAPIDefinition(oldDefinition, apiToUpdate);
+            apiProvider.saveAsyncApiDefinition(originalAPI, updateAsyncAPIDefinition);
+//            String apiDefinition = asyncApiParser.generateAsyncAPIDefinition(originalAPI);
```
Comment (dnwick) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java:365]: remove commented lines

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx [hunk 2]
```diff
@@ -197,6 +197,9 @@ export default function Resources(props) {
             }
             case 'add': {
                 const parameters = extractPathParameters(data.target, openAPISpec);
+                console.log('// ========== parameters ===========');
```
Comment (dnwick) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx:200]: remove console logs

**Review by dnwick — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParser.java [hunk 3]
```diff
@@ -1690,4 +1699,46 @@ public String getAsyncApiDefinitionForStore(API api, String asyncAPIDefinition,
...
+        Iterator<Scope> iterator = apiToUpdate.getScopes().iterator();
+        while (iterator.hasNext()) {
+            Scope scope = iterator.next();
+            scopes.put(scope.getName(), scope.getDescription());
+            scopeBindings.put(scope.getName(), scope.getRoles());
+        }
+        oauth2SecurityScheme.flows.implicit.scopes = scopes;
+
+        Extension xScopeBindings = oauth2SecurityScheme.flows.implicit.createExtension();
+        xScopeBindings.name = "x-scopes-bindings";
```
Comment (senthuran16) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParser.java:1736]: Shall we use constants for `"x-scopes-bindings"`?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/asyncapi/Runtime.jsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ */
+export default function Runtime(props) {
+    const {
+        operation, operationsDispatcher, target, verb, api,
+    } = props;
+    operation.runtime = operation.runtime || { };
+
+    const buildCallbackURL = () => {
+        return `https://{GATEWAY_HOST}:9021/${api.context.toLowerCase()}/${api.version}/`
+            + `webhooks_events_receiver_resource?topic=${target.toLowerCase()}`;
```
Comment (RakhithaRR) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/asyncapi/Runtime.jsx:41]: Shall we update this as mentioned in https://github.com/wso2/product-apim/issues/10188 ?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java (hunk 1 above)
Comment (lasanthaS) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParser.java (hunk 3 above)
Comment (lasanthaS) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/AsyncApiParser.java]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/asyncapi/Runtime.jsx (hunk 4 above)
Comment (lasanthaS) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/asyncapi/Runtime.jsx]: Fixed

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx (hunk 2 above)
Comment (lasanthaS) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/Resources.jsx]: Fixed

**Review by tmkasun — APPROVED** [review]

**Review by dnwick — APPROVED** [review]

**Comment by malinthaprasan** [comment]
Running actions in forked repo: https://github.com/malinthaprasan/carbon-apimgt/actions/runs/687669528

**Comment by malinthaprasan** [comment]
1st Run - failed: APIInvocationWithMessageTypeProperty (WebSocketAPITestCase is passed)
2nd Run (forked) - APIInvocationWithMessageTypeProperty is passed, failed WebSocketAPITestCase

**Pull request merged**